packages/@aws-cdk/aws-stepfunctions/lib/states/state.ts

```diff
@@ -579,7 +580,7 @@ export function renderJsonPath(jsonPath?: string): undefined | null | string {
   if (jsonPath === undefined) { return undefined; }
   if (jsonPath === JsonPath.DISCARD) { return null; }
```

Comment:
I think this line might still need to be updated for this to work properly.  Even though `JsonPath.DISCARD` has been updated to a JsonNull token, this line will still evaluate truthy and return null, which in other languages like Python would translate to `undefined`.  Perhaps this should return a JsonNull token as well?